include/hipSYCL/sycl/libkernel/cuda/group_functions.hpp

```diff
@@ -0,0 +1,231 @@
+/*
+ * This file is part of hipSYCL, a SYCL implementation based on CUDA/HIP
+ *
+ * Copyright (c) 2018 Aksel Alpay
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifdef SYCL_DEVICE_ONLY
+#ifdef HIPSYCL_PLATFORM_CUDA
+
+#ifndef HIPSYCL_LIBKERNEL_CUDA_GROUP_FUNCTIONS_HPP
+#define HIPSYCL_LIBKERNEL_CUDA_GROUP_FUNCTIONS_HPP
+
+namespace hipsycl {
+namespace sycl {
+
+namespace detail {
+
+template <typename T, typename std::enable_if<std::is_arithmetic<T>::value, T>::type * = nullptr>
+__device__
+T _shfl(T x, int id) {
```

Comment:
We don't have a convention for `_identifier`. There are some `__identifier()` for compiler-glue stuff or low-level macros such as `__hipsycl_lid_x` etc. How about just using `shufl_impl`?

Reply:
Do you really want `shufl_impl` or instead of `shfl_impl`/`shuffle_impl`? :)

Reply:
:laughing: You're absolutely right, let's use `shuffle_impl`.